### PR TITLE
Handle watch initial get failure a bit better.

### LINKED ISSUE #488: handle failure of the initial get in etcd watch more cleanly
`I0211 10:43:06.002665     9 continuous_fetcher.cc:112] starting fetch with tree size: 0`
`W0211 10:43:26.403292    17 tree_signer-inl.h:246] problem getting sequenced entry 0: 2: unknown error`

No apparent error in the etcd logs.

### PR DESCRIPTION
**Pull request by AlCutter**
Fixes #488

### CODE REVIEW

**Comment by pphaneuf** [comment]
Somewhat related to #811.

File: cpp/util/etcd.cc [hunk 1]
```diff
@@ -472,9 +472,18 @@ void EtcdClient::WatchInitialGetDone(WatchState* state, GetResponse* resp,
   }
 
   // TODO(pphaneuf): Need better error handling here. Have to review
-  // what the possible errors are, most of them should probably be
-  // dealt with using retries?
-  CHECK(task->status().ok()) << "initial get error: " << task->status();
+  // what the possible errors are, for now we'll just retry after a delay.
+  if (!task->status().ok()) {
+    LOG(WARNING) << "Initial get error: " << task->status();
+    LOG(INFO) << "Will retry initial get in "
```
Comment (pphaneuf) [inline cpp/util/etcd.cc]: Could you combine these into a single log message?

Comment (pphaneuf) [inline cpp/util/etcd.cc]: Any worry about a WARNING here, as well? For disk flush reasons, I mean? I forget which executor this is on... :stuck_out_tongue:

Comment (AlCutter) [inline cpp/util/etcd.cc]: Done.

**Comment by pphaneuf** [comment]
Done with my review, just a minor thing...

File: cpp/util/etcd.cc (hunk 1 above)
Comment (AlCutter) [inline cpp/util/etcd.cc]: I'm assuming it's a rare enough event that the `WARNING` is ok.

**Comment by pphaneuf** [comment]
You're based on an old `master`, as well, you should pull and rebase?

**Comment by pphaneuf** [comment]
LGTM

**Comment by pphaneuf** [comment]
Wait, oops, you need to push, I think... :smiley:

**Comment by AlCutter** [comment]
There you go, I squashed it in prep since it's a tiny change.

**Pull request merged**